pandas/tests/indexes/datetimes/test_tools.py

```diff
@@ -101,6 +101,25 @@ def test_to_datetime_format_YYYYMMDD(self, cache):
         expected = Series(["20121231", "20141231", "NaT"], dtype="M8[ns]")
         tm.assert_series_equal(result, expected)
 
+    @pytest.mark.parametrize(
+        "input_s, expected",
+        [
+            # strings with None
+            [
+                Series(["19801222", None, "20010112"]),
+                Series([Timestamp("19801222"), Timestamp(None), Timestamp("20010112")]),
+            ],
+            # Integers with None
+            [
+                Series([19801222, None, 20010112]),
+                Series([Timestamp("19801222"), Timestamp(None), Timestamp("20010112")]),
```

Comment:
since the expected is the same, just put that directly in the test function (just parameterize on input_s)

Reply:
Moved the expected into function as suggested